File: Day4&5/romanToint.java
class Solution {
    public int romanToInt(String s) {
        
        HashMap<Character ,Integer> map = new HashMap<Character,Integer>();
        map.put('I',1);
        map.put('V',5);
        map.put('X',10);
        map.put('L',50);
        map.put('C',100);
        map.put('D',500);
        map.put('M',1000);
        
        int num = 0;
        
        for (int i = 0; i < s.length(); i++){
                if (s.charAt(i) == 'V' || s.charAt(i) == 'X'){
                if(i != 0 && s.charAt(i-1) == 'I')
                    num -= 2;
                }
            
            else if (s.charAt(i) == 'L' || s.charAt(i) == 'C'){
                if(i != 0 && s.charAt(i-1) == 'X')
                    num -= 20;
                }
           
            else if (s.charAt(i) == 'D' || s.charAt(i) == 'M'){
                if(i != 0 && s.charAt(i-1) == 'C')
                    num -= 200;
                }
            num += map.get(s.charAt(i));
          
        }
        return num;
        
    }
        
    }

File: Day4&5/romanToint.cpp
class Solution {
public:
    int romanToInt(string s) {
       
        int num = 0;
        map<char,int> romans = {
            {'I',1},
            {'V',5},
            {'X',10},
            {'L',50},
            {'C',100},
            {'D',500},
            {'M',1000}
        };            
        
        for (int i = 0; i < s.length(); i++){
            
            if (s[i] == 'V' || s[i] == 'X'){
                if(i != 0 && s[i-1] == 'I')
                    num -= 2;
                }
            
            else if (s[i] == 'L' || s[i] == 'C'){
                if(i != 0 && s[i-1] == 'X')
                    num -= 20;
                }
           
            else if (s[i] == 'D' || s[i] == 'M'){
                if(i != 0 && s[i-1] == 'C')
                    num -= 200;
                }
            num += romans.at(s[i]);
          
        }
        return num;
        
    }
};
